client/state/analytics/middleware.js

```diff
@@ -0,0 +1,40 @@
+import analytics from 'analytics';
+import invoke from 'lodash/invoke';
+import reduxAnalytics from 'redux-analytics';
+
+import {
+	BUMP_STAT,
+	RECORD_EVENT,
+	RECORD_PAGE_VIEW
+} from './constants';
+
+const eventServices = {
+	ga: ( { category, action, label, value } ) => analytics.ga.recordEvent( category, action, label, value ),
+	tracks: ( { name, properties } ) => analytics.tracks.recordEvent( name, properties )
+};
+
+const pageViewServices = {
+	ga: ( { url, title } ) => analytics.ga.recordPageView( url, title ),
+	default: ( { url, title } ) => analytics.pageView.record( url, title )
+};
+
+const statBump = ( { group, name } ) => analytics.mc.bumpStat( group, name );
+
+export const dispatcher = ( { analytics: { type, payload } } ) => {
+	const { service = 'default' } = payload;
+
+	switch ( type ) {
+		case RECORD_EVENT:
+			return invoke( eventServices, service, payload );
+
+		case RECORD_PAGE_VIEW:
+			return invoke( pageViewServices, service, payload );
+
+		case BUMP_STAT:
+			return statBump( payload );
+	}
+};
+
+export const analyticsMiddleware = reduxAnalytics( dispatcher );
```

Comment:
@aduth this is almost the same way I feel about it. I'm going to let a few more comments trickle in. If nobody has more of an opinion, I'll rewrite without the middleware. As far as I know, the theme middleware is the only other place using that dependency.

It's my thought that the theme middleware could disappear after this gets merged because we could rewrite those calls using the `withAnalytics()` wrapper, though I'm not sure how extensively that's used.